src/managers/ProductManager/ProductManager.ts

```diff
@@ -0,0 +1,173 @@
+  ) => {
+    for (const obj of this.productsValuesToCheck) {
+      const { key, selector, shouldDelete } = obj;
+
+      const value = productValues[key];
+
+      if (shouldDelete(value)) {
+        element.querySelector(selector)?.remove();
+      }
+    }
+
+    return element;
+  };
+
+  private deleteExistingProduct = (id: string) => {
+    const productIdSelector = getProductsIdSelectorIfExists(this.page);
+    const productIdExtractorKey = getProductsIdExtractorIfExists(this.page);
+
+    Array.from(
+      this.productsContainer.querySelectorAll(productIdSelector),
+    ).forEach((product) => {
+      const doesContainId =
+        product instanceof HTMLElement &&
+        productIdExtractorKey in product.dataset;
+
+      if (!doesContainId) return;
+
+      if (product.dataset[productIdExtractorKey] !== id) return;
+
+      if (product.classList.contains('product')) {
+        product.remove();
+      } else {
+        product.parentElement?.remove();
+      }
+    });
+  };
+
+  private addTagToProductElement = (
+    productElement: Element,
+    selector: string,
+    dsaUrl: string | undefined,
+  ) => {
+    const labelElementContainer = document.createElement('p');
+    labelElementContainer.classList.add(TAG_STYLES_CLASS);
+    labelElementContainer.classList.add('product__name');
+
+    const labelElement = document.createElement('span');
+    labelElement.textContent = window.sponsoredProductConfig.tagLabel;
+    labelElementContainer.appendChild(labelElement);
+
+    if (dsaUrl) {
+      const sponsoredLabelLink = document.createElement('a');
+      sponsoredLabelLink.style.marginLeft = '4px';
+      sponsoredLabelLink.style.textDecoration = 'none';
+      sponsoredLabelLink.style.pointerEvents = 'auto';
+      sponsoredLabelLink.href = dsaUrl;
+      sponsoredLabelLink.target = '_blank';
+      sponsoredLabelLink.innerHTML = DSA_ICON;
+
+      labelElementContainer.appendChild(sponsoredLabelLink);
+    }
+
+    productElement.querySelector(selector)?.prepend(labelElementContainer);
+
+    return productElement;
+  };
+
+  private injectTagStyles = (styles: string) => {
+    const stylesElement = document.createElement('style');
+    stylesElement.innerHTML = styles;
+    this.productsContainer.appendChild(stylesElement);
+  };
+
+  public injectProduct = (products: TFormattedProduct[]) => {
+    let productsContainer = this.productsContainer;
+    const tagStyles = TAGS_STYLES[this.currentTemplate];
+
+    this.injectTagStyles(tagStyles.styles);
+
+    if (this.isSlider) {
+      // override productsContainer to be the actual container of slider items
+      productsContainer = this.productsContainer.querySelector(
+        '.slick-track',
+      ) as HTMLDivElement;
+    }
+
+    const { productsCount } =
+      window.sponsoredProductConfig[mapConfigPages(this.page)];
+
+    for (const [index, product] of products.entries()) {
+      if (index >= productsCount) break;
```

Comment:
Bardzo fajnie, że można sterować ilością wyświetlanych produktów, ale powinno to działać troszkę inaczej.
Komentarz w AdManger